pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/AuthorizationService.java

```diff
@@ -273,7 +281,7 @@ public CompletableFuture<Boolean> canLookupAsync(TopicName topicName, String rol
             if (isSuperUser) {
                 return CompletableFuture.completedFuture(true);
             } else {
-                return provider.canLookupAsync(topicName, role, authenticationData);
+                return provider.allowTopicOperationAsync(topicName, role, TopicOperation.LOOKUP, authenticationData);
```

Comment:
This is only tangentially related to your PR, but it seems like a broken abstraction that the `AuthorizationService` decides that the `superuser` privilege supersedes the `AuthenticationProvider` implementaiton. Instead, I think we could just call `provider.allowTopicOperationAsync(topicName, role, TopicOperation.LOOKUP, authenticationData);`, as you've done here.

My goal is to move that logic into the provider so that plugin implementations do not need to deal with the complexity of when to check for super user and when not to. All of this logic should be delegated to the provider, in my opinion.

That change is probably worth its own PR, though.